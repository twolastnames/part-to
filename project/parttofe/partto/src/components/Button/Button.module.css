.buttonBase {
  color: green;
  font-size: 30px;
  position: relative;
  top: 150px;
}
