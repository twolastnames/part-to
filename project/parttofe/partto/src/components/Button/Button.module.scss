.container {
  height: 36px;
  width: 36px;
}

.button {
  border-radius: 36px;
  height: 36px;
  width: 36px;
}
