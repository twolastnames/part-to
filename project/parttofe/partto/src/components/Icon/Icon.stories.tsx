import React from "react";
import type { Meta, StoryObj } from "@storybook/react";

import { Icon, Next } from "./Icon";
import { ShellProvider } from "../../providers/ShellProvider";
import { Size } from "./IconTypes";

const meta: Meta<typeof Icon> = {
  component: Icon,
  decorators: (Story) => (
    <ShellProvider>
      <Story />
    </ShellProvider>
  ),
};
export default meta;

type Story = StoryObj<typeof Icon>;

export const Medium: Story = {
  args: { definition: Next },
};

export const Small: Story = {
  args: { definition: Next, size: Size.Small },
};

export const Large: Story = {
  args: { definition: Next, size: Size.Large },
};

export const ExtraLarge: Story = {
  args: { definition: Next, size: Size.ExtraLarge },
};
