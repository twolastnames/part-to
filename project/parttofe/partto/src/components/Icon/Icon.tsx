import React, { ForwardRefExoticComponent, RefAttributes } from "react";
import { rem } from "@mantine/core";
import {
  Icon as TablerIcon,
  IconProps as TablerIconProps,
  IconPlayerTrackNext,
  IconPlayerPlay,
  IconList,
  IconFile,
  IconCaretDown,
  IconCaretUp,
  IconChefHat,
  IconCooker,
  IconToolsKitchen,
} from "@tabler/icons-react";

import classes from "./Icon.module.scss";

export type IconType = ForwardRefExoticComponent<
  TablerIconProps & RefAttributes<TablerIcon>
>;

export enum Size {
  Small = 20,
  Medium = 30,
  Large = 50,
  ExtraLarge = 80,
}

export interface IconProps {
  definition: IconType;
  size?: Size;
  onClick?: () => void;
}

export function Icon({ onClick, definition, size }: IconProps) {
  return React.createElement(definition, {
    ...(onClick
      ? {
          onClick,
          className: classes.clickable,
        }
      : {}),
    style: {
      width: rem((size || Size.Medium) as number),
      height: rem((size || Size.Medium) as number),
    },
  });
}

export const Next = IconPlayerTrackNext;
export const Start = IconPlayerPlay;
export const List = IconList;
export const File = IconFile;
export const Up = IconCaretUp;
export const Down = IconCaretDown;
export const ChefHat = IconChefHat;
export const Oven = IconCooker;
export const KitchenTools = IconToolsKitchen;
