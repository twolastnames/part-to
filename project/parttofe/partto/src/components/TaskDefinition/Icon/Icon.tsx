import React from "react";

import classes from "./Icon.module.scss";
import { ClassNames, IconProps } from "./IconTypes";
import {
  Duty as DutyIcon,
  Imminent as ImminentIcon,
  Icon as PhysicalIcon,
  Task as TaskIcon,
} from "../../Icon/Icon";

export const Duty: ClassNames = {
  imminent: classes.hidden,
  duty: classes.icon,
  task: classes.hidden,
};

export const Task: ClassNames = {
  imminent: classes.hidden,
  duty: classes.hidden,
  task: classes.icon,
};

export const Imminent: ClassNames = {
  imminent: classes.icon,
  duty: classes.hidden,
  task: classes.hidden,
};

export function Icon({ classNames: { task, duty, imminent } }: IconProps) {
  return (
    <>
      <span className={imminent}>
        <PhysicalIcon definition={ImminentIcon} />
      </span>
      <span className={duty}>
        <PhysicalIcon definition={DutyIcon} />
      </span>
      <span className={task}>
        <PhysicalIcon definition={TaskIcon} />
      </span>
    </>
  );
}
