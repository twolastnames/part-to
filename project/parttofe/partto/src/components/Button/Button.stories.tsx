import React from "react";
import type { Meta, StoryObj } from "@storybook/react";

import { Button } from "./Button";
import { ShellProvider } from "../../ShellProvider";
import { Next, Start } from "../Icon/Icon";

const meta: Meta<typeof Button> = {
  component: Button,
  decorators: (Story) => (
    <ShellProvider>
      <Story />
    </ShellProvider>
  ),
};
export default meta;

type Story = StoryObj<typeof Button>;

export const NextButton: Story = {
  args: {
    onClick: () => console.log("clicked"),
    icon: Next,
    text: "Is Complete",
  },
};

export const StartButton: Story = {
  args: {
    onClick: () => console.log("clicked"),
    icon: Start,
    text: "Start Making Dinner",
  },
};
