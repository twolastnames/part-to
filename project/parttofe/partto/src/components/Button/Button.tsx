import React from "react";
import { ActionIcon as MantineButton, Tooltip } from "@mantine/core";

import classes from "./Button.module.scss";
import { IconType, Icon, Size as IconSize } from "../Icon/Icon";

export interface ButtonProps {
  text: string;
  icon: IconType;
  onClick: () => void;
}

export const Button = ({ onClick, icon, text }: ButtonProps) => (
  <>
    <Tooltip label={text} data-testid="Button">
      <div onClick={onClick} className={classes.container}>
        <MantineButton
          data-testid="Button"
          aria-label={text}
          className={classes.button}
        >
          <Icon definition={icon} size={IconSize.Small} />
        </MantineButton>
      </div>
    </Tooltip>
  </>
);
