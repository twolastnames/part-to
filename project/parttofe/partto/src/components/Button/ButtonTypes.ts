import { IconType } from "../Icon/IconTypes";

export interface ButtonProps {
  text: string;
  icon: IconType;
  onClick: () => void;
}
