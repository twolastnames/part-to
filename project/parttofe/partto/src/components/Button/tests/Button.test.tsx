import React from "react";
import { describe, expect, test } from "@jest/globals";
import { render, screen } from "@testing-library/react";
import { Button } from "../Button";
import { MantineProvider } from "@mantine/core";
import { Next } from "../../Icon/Icon";

describe("Button", () => {
  beforeAll(() => {});

  test("snapshot", () => {
    render(
      <MantineProvider>
        <Button icon={Next} text={"Hello"} onClick={() => undefined} />
      </MantineProvider>,
    );
    const component = screen.getByTestId("Button");
    expect(component).toMatchSnapshot();
  });
});
