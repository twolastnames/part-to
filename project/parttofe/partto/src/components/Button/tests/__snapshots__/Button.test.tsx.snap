// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot 1`] = `
<button
  aria-label="Hello"
  class="mantine-focus-auto mantine-active button m_8d3f4000 mantine-ActionIcon-root m_87cf2631 mantine-UnstyledButton-root"
  data-testid="Button"
  style="--ai-color: var(--mantine-color-white);"
  type="button"
>
  <span
    class="m_8d3afb97 mantine-ActionIcon-icon"
  >
    <svg
      class="tabler-icon tabler-icon-player-track-next "
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M3 5v14l8 -7z"
      />
      <path
        d="M14 5v14l8 -7z"
      />
    </svg>
  </span>
</button>
`;
